awx/main/models/inventory.py

```diff
@@ -1797,3 +1815,726 @@ class Meta:
+            cred_kind = inventory_update.source.replace('ec2', 'aws')
+            if cred_kind in dir(builtin_injectors):
+                getattr(builtin_injectors, cred_kind)(credential, injected_env, private_data_dir)
+                if safe:
+                    from awx.main.models.credential import build_safe_env
+                    return build_safe_env(injected_env)
+        elif self.base_injector == 'template':
+            injected_env['INVENTORY_UPDATE_ID'] = str(inventory_update.pk)  # so injector knows this is inventory
+            safe_env = injected_env.copy()
+            args = []
+            safe_args = []
+            credential.credential_type.inject_credential(
+                credential, injected_env, safe_env, args, safe_args, private_data_dir
+            )
+            if safe:
+                return safe_env
+        return injected_env
+
+    def get_plugin_env(self, inventory_update, private_data_dir, private_data_files, safe=False):
+        return self._get_shared_env(inventory_update, private_data_dir, private_data_files, safe)
+
+    def get_script_env(self, inventory_update, private_data_dir, private_data_files, safe=False):
+        injected_env = self._get_shared_env(inventory_update, private_data_dir, private_data_files, safe)
+
+        # Put in env var reference to private ini data files, if relevant
+        if self.ini_env_reference:
+            credential = inventory_update.get_cloud_credential()
+            cred_data = private_data_files.get('credentials', '')
+            injected_env[self.ini_env_reference] = cred_data[credential]
+
+        return injected_env
+
+    def build_private_data(self, inventory_update, private_data_dir):
+        if self.should_use_plugin():
+            return self.build_plugin_private_data(inventory_update, private_data_dir)
+        else:
+            return self.build_script_private_data(inventory_update, private_data_dir)
+
+    def build_script_private_data(self, inventory_update, private_data_dir):
+        return None
+
+    def build_plugin_private_data(self, inventory_update, private_data_dir):
+        return None
+
+    @staticmethod
+    def dump_cp(cp, credential):
+        """Dump config parser data and return it as a string.
+        Helper method intended for use by build_script_private_data
+        """
+        if cp.sections():
+            f = StringIO()
+            cp.write(f)
+            private_data = private_data = {'credentials': {}}
+            private_data['credentials'][credential] = f.getvalue()
+            return private_data
+        else:
+            return None
+
+
+class azure_rm(PluginFileInjector):
+    plugin_name = 'azure_rm'
+    # FIXME: put in correct version when Ansible core fixes are in
+    # 2.8 does not have all needed hostvars https://github.com/ansible/ansible/issues/51864
+    # also affected by unsafe group names issue
+    # initial_version = '2.8'
```

Comment:
so right now it does not have an initial version --> that means that `should_use_plugin()` will always return`False` at this point, correct?

Reply:
yes.

So all plugin logic like that is non-functional now - they are only used by the unit tests. But we're still talking about the flag for enablement, which could change that.